packages/flame_forge2d/example/lib/composition_sample.dart

```diff
@@ -0,0 +1,61 @@
+import 'package:flame/components.dart';
+import 'package:flame/palette.dart';
+import 'package:flame_forge2d/flame_forge2d.dart';
+import 'package:flame_forge2d/forge2d_game.dart';
+import 'package:forge2d/forge2d.dart';
+
+import 'balls.dart';
+import 'boundaries.dart';
+
+class CompositionSample extends Forge2DGame with HasTappableComponents {
+  static const info = '''
+This example shows how to compose a `BodyComponent` together with a normal Flame
+component. Click the ball to see the number increment.
+''';
+
+  CompositionSample() : super(zoom: 20, gravity: Vector2(0, -10.0));
+
+  @override
+  Future<void> onLoad() async {
+    final boundaries = createBoundaries(this);
+    boundaries.forEach(add);
+    final center = screenToWorld(viewport.effectiveSize / 2);
+    add(TapableBall(center));
+  }
+}
+
+class TapableBall extends Ball with Tappable {
+  late final TextComponent textComponent;
+  int counter = 0;
+  final TextPaintConfig _textConfig =
+      TextPaintConfig(color: BasicPalette.white.color, fontSize: 4);
+  late final TextPaint _textPaint;
+
+  TapableBall(Vector2 position) : super(position) {
+    originalPaint = BasicPalette.white.paint();
+    paint = originalPaint;
+    _textPaint = TextPaint(config: _textConfig);
+    textComponent = TextComponent(counter.toString(), textRenderer: _textPaint);
```

Comment:
I would move this to the onLoad method.

In your case, it will work fine on the constructor, but given the lifecycle of the flame components, it is safe to initialize things on the onLoad, as we guarantee that everything will be set up on the component by the time the method is called.

Reply:
Good point, will change it!